include/fmt/format.h

```diff
@@ -295,12 +295,22 @@ FMT_CONSTEXPR20 auto bit_cast(const From& from) -> To {
   return to;
 }
 
+#ifdef _WIN32
+constexpr auto is_big_endian() -> bool { return false; }
+#elif defined(__BIG_ENDIAN__)
+constexpr auto is_big_endian() -> bool { return true; }
+#elif defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__)
+constexpr auto is_big_endian() -> bool {
+  return __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__;
+}
+#else
```

Comment:
I suggest moving the ifdefs into the body of `is_big_endian` to avoid repeating its signature. I don't think we care about `constexpr` here, it will be inlined and optimized away.

Reply:
Done